src/scenes/ProgressReports/EditForm/Steps/ProgressStep/ProgressStep.tsx

```diff
@@ -0,0 +1,239 @@
+import { useMutation } from '@apollo/client';
+import {
+  Box,
+  Grid,
+  ToggleButton,
+  ToggleButtonGroup,
+  Typography,
+} from '@mui/material';
+import { groupBy } from 'lodash';
+import { useState } from 'react';
+import { useDialog } from '~/components/Dialog';
+import {
+  StepProgressFragment,
+  UpdateStepProgressDocument,
+} from '../../../../Products/Detail/Progress/ProductProgress.graphql';
+import { ProgressSummaryCard } from '../../../../ProgressReports/Detail/ProgressSummaryCard';
+import { UpdatePeriodicReportDialog } from '../../../../Projects/Reports/UpdatePeriodicReportDialog';
+import { ProductTable } from '../../../Detail/ProductTable';
+import { ProgressReportCard } from '../../../Detail/ProgressReportCard';
+import { colorPalette } from '../../colorPalette';
+import { useProgressReportContext } from '../../ProgressReportContext';
+import {
+  ProgressReportEditFragment,
+  ProgressReportProgressFragment,
+} from '../../ProgressReportEdit.graphql';
+import { RoleIcon } from '../../RoleIcon';
+
+const transparentBgSx = {
+  backgroundColor: 'transparent',
+};
+
+const iconSx = {
+  height: 36,
+  width: 36,
+  mr: 0,
+};
+
+const ToggleButtonSx = (role: string) => ({
+  p: 0.25,
+  pr: 1,
+  mr: 0,
+  borderRadius: 0.6,
+  '&.Mui-selected': {
+    backgroundColor: colorPalette.stepperCard.iconBackground[role],
+  },
+  '&.Mui-selected:hover': {
+    backgroundColor: colorPalette.stepperCard.iconBackground[role],
+  },
+});
+
+const variants = ['partner', 'official'] as const;
```

Comment:
These variants are designed to be dynamic so that additional variants can be added in future. That's why they come from the schema/API as strings instead of an enum. We shouldn't assume what those "enum" values are going to be. We should be able to handle a change in this list dynamically based on data. So as more of an exception to our normal rule, I'd just use the `string` type here.

This list should be dynamic from each API call. Each user could have a subset of these variants available (0, 1, 2+).
So we need to handle those first two cases as well. I'd avoid the tabs all together if there's only one variant available, like for the Field Partner where they can just see their own input.
If there's zero we should display a no permission error message.